src/components/productDetail/ProductDetail.component.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { CartIcon } from '@/assets/icons/Cart.tsx';
+import { LeftArrowIcon } from '@/assets/icons/LeftArrow.tsx';
+import { RightArrowIcon } from '@/assets/icons/RightArrow.tsx';
+import { DesktopOnly, MobileOnly } from '@/components/media-helpers/MediaHelpers.tsx';
+import { useCart } from '@/contexts/CartContextProvider.tsx';
+import type { Product } from '@/types/interfaces/Product.ts';
+import { fetchProduct } from '@/utils/fetchApi.ts';
+
+import styles from './productDetail.module.css';
+
+export const ProductDetail: React.FC = () => {
+    const { id } = useParams();
+    const [product, setProduct] = useState<Product>();
+    const [mainImage, setMainImage] = useState('');
+
+    const cart = useCart();
+
+    const handleCartClick = () => {
+        if (product) {
+            cart.toggleItem(product.id);
+        }
+    };
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if (id) {
+            const fetchedProductById = async () => {
+                const fetchedProduct = await fetchProduct(id);
+                setProduct(fetchedProduct);
+                setMainImage(fetchedProduct?.images[0]);
+            };
+
+            fetchedProductById();
+        }
+    }, [id]);
+
+    const handleBackClick = () => {
+        navigate(-1);
+    };
+
+    const handleThumbnailClick = (image: string) => {
+        setMainImage(image);
+    };
+
+    const handleNextImage = () => {
+        if (product) {
+            const currentIndex = product.images.indexOf(mainImage);
+            const nextIndex = (currentIndex + 1) % product.images.length;
+            setMainImage(product.images[nextIndex]);
+        }
+    };
+
+    const handlePreviousImage = () => {
+        if (product) {
+            const currentIndex = product.images.indexOf(mainImage);
+            const previousIndex = (currentIndex - 1 + product.images.length) % product.images.length;
+            setMainImage(product.images[previousIndex]);
+        }
+    };
+
+    return (
+        <div className={styles.main}>
+            <div className={styles.detailWrapper}>
+                <div className={styles.photosWrapper}>
+                    <div className={styles.thumbnailWrapper}>
+                        {product?.images.map((item, index) => (
```

Comment:
instead of adding `?` sign to `product` display whole template if product exists